src/Microsoft.Azure.SignalR/ClientInvocation/CallerClientResultsManager.cs

```diff
@@ -20,34 +23,54 @@ internal sealed class CallerClientResultsManager : ICallerClientResultsManager,
         private long _lastInvocationId = 0;
 
         private readonly IHubProtocolResolver _hubProtocolResolver;
+        private IEndpointRouter? _endpointRouter { get; }
+        private IServiceEndpointManager? _serviceEndpointManager { get; }
+        private readonly AckHandler _ackHandler = new();
 
-        public CallerClientResultsManager(IHubProtocolResolver hubProtocolResolver)
+        public CallerClientResultsManager(IHubProtocolResolver hubProtocolResolver, IServiceEndpointManager? serviceEndpointManager = null, IEndpointRouter? endpointRouter = null)
         {
             _hubProtocolResolver = hubProtocolResolver ?? throw new ArgumentNullException(nameof(hubProtocolResolver));
+            if (serviceEndpointManager == null && endpointRouter != null) throw new ArgumentNullException(nameof(serviceEndpointManager));
+            if (serviceEndpointManager != null && endpointRouter == null) throw new ArgumentNullException(nameof(endpointRouter));
+            _serviceEndpointManager = serviceEndpointManager;
+            _endpointRouter = endpointRouter;
+            
         }
 
         public string GenerateInvocationId(string connectionId)
         {
             return $"{connectionId}-{_clientResultManagerId}-{Interlocked.Increment(ref _lastInvocationId)}";
         }
 
-        public Task<T> AddInvocation<T>(string connectionId, string invocationId, CancellationToken cancellationToken)
+        public Task<T> AddInvocation<T>(string hub, string connectionId, string invocationId, CancellationToken cancellationToken)
         {
             var tcs = new TaskCompletionSourceWithCancellation<T>(
                 cancellationToken,
                 () => TryCompleteResult(connectionId, CompletionMessage.WithError(invocationId, "Canceled")));
 
+            var ackNumber = 1;
+            if (_serviceEndpointManager != null && _endpointRouter != null) {
+                var serviceEndpoints = _serviceEndpointManager.GetEndpoints(hub);
+                ackNumber = _endpointRouter.GetEndpointsForConnection(connectionId, serviceEndpoints).Count();
```

Comment:
This should be same as the one called in [MultiEndpointServiceConnectionContainer.cs](https://github.com/Azure/azure-signalr/pull/1852/files#diff-1f321e250ff8676cbfb17272b420c048bc70705b398a2e315fcbe28c76cb2f57)